### PR TITLE
samples: Bluetooth: Add configuration file for all controller features

### PR DESCRIPTION
**Pull request by cvinayak**
Fix missing includes and rearrange header includes to be able
to build a full feature set Controller.

Add project configuration overlay file to build the current
full feature set Bluetooth Low Energy Controller.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

File: samples/bluetooth/hci_uart/overlay-all.conf [hunk 1]
```diff
@@ -0,0 +1,69 @@
+CONFIG_BT_BUF_EVT_RX_COUNT=16
```
Comment (Thalley) [inline samples/bluetooth/hci_uart/overlay-all.conf:1]: Just out of curiosity: What does `all` refer her to here? All controller features?

Can we ensure that this is true, even when new features are added?

File: samples/bluetooth/hci_uart/overlay-all.conf [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+CONFIG_BT_CTLR_ADVANCED_FEATURES=y
+CONFIG_BT_CTLR_ADV_AUX_SET=3
+CONFIG_BT_CTLR_ADV_AUX_PDU_BACK2BACK=y
+CONFIG_BT_CTLR_ADV_SYNC_SET=3
+CONFIG_BT_CTLR_ADV_SYNC_PDU_BACK2BACK=y
+CONFIG_BT_CTLR_ADV_DATA_BUF_MAX=6
+
+# Increase the below to receive interleaved advertising chains
+# CONFIG_BT_CTLR_SCAN_AUX_SET=2
```
Comment (Thalley) [inline samples/bluetooth/hci_uart/overlay-all.conf:49]: Why is this commented out?

File: samples/bluetooth/hci_uart/overlay-all.conf (hunk 1 above)
Comment (rugeGerritsen) [reply samples/bluetooth/hci_uart/overlay-all.conf:1]: We could at consider naming this file `overlay-all-bt_ll_sw_split.conf` as it looks like it contains controller specific configurations

Comment (cvinayak) [reply samples/bluetooth/hci_uart/overlay-all.conf]: > Just out of curiosity: What does `all` refer her to here? All controller features?
> 
> Can we ensure that this is true, even when new features are added?

All conf files are manual process, updated by contributors. No, conf files can not be used to ensure new features are added. Atleast this file will be build tested in CI and will catch any config be removed.

I just followed https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/subsys/shell/shell/shell_min_cmds_all.conf, i assume is updated manually when adding new features.

Comment (cvinayak) [reply samples/bluetooth/hci_uart/overlay-all.conf]: > We could at consider naming this file `overlay-all-bt_ll_sw_split.conf` as it looks like it contains controller specific configurations

done.

File: samples/bluetooth/hci_uart/overlay-all.conf (hunk 2 above)
Comment (cvinayak) [reply samples/bluetooth/hci_uart/overlay-all.conf]: Removed comment and set it to default value.

File: samples/bluetooth/hci_uart/overlay-all.conf (hunk 1 above)
Comment (Thalley) [reply samples/bluetooth/hci_uart/overlay-all.conf]: > All conf files are manual process, updated by contributors. No, conf files can not be used to ensure new features are added. Atleast this file will be build tested in CI and will catch any config be removed.

IIRC when adding the new LLCP, there was a script to generate all possible mutations of several Kconfigs for testing building them. 

In theory it is possible to recreate something similar for this, where all Kconfigs that are `default n` to be enabled, so it should be possible. 

What the goal of this PR? Just to build a sample with them, or is it to actually run the sample with them as well?

**Review by Thalley — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**